src/org/loklak/data/ElasticsearchClient.java

```diff
@@ -988,4 +990,64 @@ public TermsBuilder getClassifierAggregation(String classifierName) {
                 AggregationBuilders.sum("sum_probability").field(probabilityField)
             );
     }
+
+    public HashMap<String, HashMap<String, Double>> classifierScore(String index, String classifierName, ArrayList<String> classes, String startDate, String endDate) {
+        if (startDate == null && endDate == null) {
+            return classifierScore(index, classifierName, classes);
+        }
+
+        SearchRequestBuilder request = elasticsearchClient.prepareSearch(index)
+            .setSearchType(SearchType.QUERY_THEN_FETCH)
+            .setQuery(QueryBuilders.matchAllQuery())
+            .setFrom(0)
+            .setSize(0)
+            .addAggregation(getClassifierAggregation(classifierName, startDate, endDate));
+
+        SearchResponse response = request.execute().actionGet();
+        Range aggrs = response.getAggregations().get("by_date");
+
+        HashMap<String, HashMap<String, Double>> aggregatedMap = new HashMap<>();
+
+        for (Range.Bucket bkt : aggrs.getBuckets()) {
+            Terms termAggrs = bkt.getAggregations().get("by_class");
+            for (Terms.Bucket bucket: termAggrs.getBuckets()) {
+                String key = bucket.getKeyAsString();
+                if (!classes.contains(key)) {
+                    continue;
+                }
+                long docCount = bucket.getDocCount();
+                Sum sum = bucket.getAggregations().get("sum_probability");
+                Avg avg = bucket.getAggregations().get("avg_probability");
+                HashMap<String, Double> map = new HashMap<>();
+                map.put("count", (double) docCount);
+                map.put("sum", sum.getValue());
+                map.put("avg", avg.getValue());
+                aggregatedMap.put(key,  map);
+            }
+        }
+        return aggregatedMap;
+    }
+
+    public AggregationBuilder getClassifierAggregation(String classifierName, String fromDate, String toDate) {
+        String probabilityField = classifierName + "_probability";
+        DateRangeBuilder db = AggregationBuilders.dateRange("by_date").field("created_at");
```

Comment:
if the variables give some hint what it does would be better like filterData, etc (but not too big variables). Please see at other places too.